upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -541,6 +541,12 @@ func (c *awsCloudImplementation) DeleteInstance(i *cloudinstances.CloudInstance)
 	return deleteInstance(c, i)
 }
 
+// DeregisterInstance drains a cloud instance and loadbalancers.
+func (c *awsCloudImplementation) DeregisterInstance(i *cloudinstances.CloudInstance) error {
```

Comment:
AWS actually has a `deregisterInstance` that does the same time. It is not exposed thought this interface though, but is called from `deleteInstance`. Think we should consolidate here.

Reply:
if I delete `deregisterInstance` from `deleteInstance` function, it means that in case of warmpool https://github.com/kubernetes/kops/blob/e0d22ce2f7700ed9be7af619ff5f029bc9d103dd/pkg/instancegroups/instancegroups.go#L133 we are not detaching lbs. Second option is that I leave `deregisterInstance` to `deleteInstance` function. In that case other than warmpool instances will get deregisterInstance twice. Which one we should use?

Reply:
actually third option is that we do not implement new `DeregisterInstance` to AWS at all, we leave it as-is which will work fine for AWS

Reply:
Can the DeregisterInstance call also be called from `DeleteSingleInstance`?
I am not sure why warmpool matters here. Those instances have not joined the cluster or really the ASGs and should be deregistered at all. Or am I missing something?

Reply:
ah right, I did not know what means warmpool